packages/x-adapter/src/schemas/types.ts

```diff
@@ -53,45 +54,48 @@ export type Schema<Source = any, Target = any> = {
  * @param OriginalSchema - The {@link Schema | schema} that will be mutable.
  * @public
  */
-export type MutableSchema<OriginalSchema extends Schema> = OriginalSchema & {
+export type MutableSchema<Source, Target> = Schema<Source, Target> & {
   /**
    * Replaces all usages of the original {@link Schema | schema} with the given one.
    *
    * @param newSchema - The {@link Schema | schema} to use instead of the original one.
    * @returns The new {@link Schema | schema} that will be used.
    */
-  $replace: <Source, Target>(
-    newSchema: Schema<Source, Target>
-  ) => MutableSchema<Schema<Source, Target>>;
+  $replace<NewSource, NewTarget>(
```

Comment:
By changing to this signature, the `this` type is correctly inferred, and we don't allow to destructure the functions.